inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp

```diff
@@ -0,0 +1,104 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <vpu/frontend/frontend.hpp>
+
+#include <ngraph/op/non_max_suppression.hpp>
+
+#include <memory>
+#include <set>
+
+namespace vpu {
+
+namespace {
+
+class StaticShapeNMS final : public StageNode {
+private:
+    StagePtr cloneImpl() const override {
+        return std::make_shared<StaticShapeNMS>(*this);
+    }
+
+    void propagateDataOrderImpl(StageDataInfo<DimsOrder>& orderInfo) override {
+    }
+
+    void getDataStridesRequirementsImpl(StageDataInfo<StridesRequirement>& stridesInfo) override {
+    }
+
+    void finalizeDataLayoutImpl() override {
+    }
+
+    void getBatchSupportInfoImpl(StageDataInfo<BatchSupport>& batchInfo) override {
+    }
+
+    StageSHAVEsRequirements getSHAVEsRequirementsImpl() const override {
+        return StageSHAVEsRequirements::OnlyOne;
+    }
+
+    void initialCheckImpl() const override {
+        assertInputsOutputsTypes(this,
+                                 {{DataType::FP16},
+                                  {DataType::FP16},
+                                  {DataType::S32},
+                                  {DataType::FP16},
+                                  {DataType::FP16}},
+                                 {{DataType::S32},
+                                  {DataType::S32}});
+    }
+
+    void finalCheckImpl() const override {
+    }
```

Comment:
While this place definitely needs refactoring - please check the same invariants at the end as well

Reply:
Fixed.